service/history/workflowExecutionUtil.go

```diff
@@ -25,6 +25,74 @@ import (
 	"github.com/uber/cadence/common"
 )
 
+type workflowContext interface {
+	getContext() workflowExecutionContext
+	getMutableState() mutableState
+	reloadMutableState() (mutableState, error)
+	getReleaseFn() releaseWorkflowExecutionFunc
+	getRunID() string
+}
+
+type workflowContextImpl struct {
+	context      workflowExecutionContext
+	mutableState mutableState
+	releaseFn    releaseWorkflowExecutionFunc
+}
+
+type updateWorkflowAction struct {
+	noop           bool
+	createDecision bool
+}
+
+var (
+	emptyUpdateAction             = updateWorkflowAction{}
```

Comment:
emptyUpdateAction is the same as updateWorkflowWithoutDecision?

Reply:
just for better reading.

Reply:
then can you rename it? emptyUpdateAction is misleading

Reply:
After discussion, I change it back to nil and update post action as pointer.